Exiled.API/Extensions/RoomExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="RoomExtensions.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.API.Extensions
+{
+    using System.Collections.Generic;
+    using Exiled.API.Enums;
+
+    /// <summary>
+    /// A set of extensions for <see cref="RoomType"/>.
+    /// </summary>
+    public static class RoomExtensions
+    {
+        /// <summary>
+        /// Checks if a <see cref="RoomType">room type</see> is a gate.
+        /// </summary>
+        /// <param name="room">The room to be checked.</param>
+        /// <returns>Returns whether the <see cref="RoomType"/> is a gate or not.</returns>
+        public static bool IsGate(this RoomType room) => room is RoomType.EzGateA or RoomType.EzGateB;
+
+        /// <summary>
+        /// Checks if a <see cref="RoomType">room type</see> is a checkpoint.
+        /// </summary>
+        /// <param name="room">The room to be checked.</param>
+        /// <returns>Returns whether the <see cref="RoomType"/> is a checkpoint or not.</returns>
+        public static bool IsCheckpoint(this RoomType room) => room is RoomType.LczChkpA
```

Comment:
Adding a bool if player want to take RoomType.HczChkp because it's don't have the checkpoints door and people can be confuse with it

Reply:
I can't understand your sentence, hence I'm out 🙃

Reply:
It's because it's checkpoint should give true only on room with the checkpoint door and the HczChkpt don't have this checkpoint door